collections/_posts/2024-10-20-tabular-explainability.md

```diff
@@ -0,0 +1,199 @@
+* ***NI - Network Intrusion Dataset*** - Classification between bad connections (intrusions or attacks) and good connections. 1,000,000 samples were used for training and 75,000 for validation. 4 groups of features (basic, content, traffic, host) were used
+* ***RW - Real-World Dataset*** - Binary classification problem. Tens of thousands of samples were used for training and validation. 8 groups of features were used. They had limited access to this private dataset and were unable to produce the same set of experiments as for the CT and NI dataset
+
+&nbsp;
+
+## Implementations
+
+* ***Teacher network***: $$N=2$$ (number of blocs), $$h=4$$ (number of heads per bloc), $$d=64$$ (token size)
+* ***Student network***: $$M=4$$ (number of blocs), $$h=1$$ (number of heads per bloc), $$d=64$$ (token size)
+* Batch size of 128 and Adam optimizer
+* $$\lambda_{CT} = 0.005$$, $$\lambda_{NI} = 0.01$$, and $$\lambda_{RW} = 0.9$$
+* Each experiment was repeated five times for each CT and NI student and ten times for each RW student
+
+&nbsp;
+
+## Teacher/student classification performances
+
+* Comparaison with two standard machine learning methods: LightGBM and XGBoost
+* Conceptual transformer is the student network
+* The main goal is not to obtain the most performant model, but rather a comparable model that is better suited for explainability purposes
+
+<div style="text-align:center">
+<img src="/collections/images/tabular_explainability/results_teacher_performances.jpg" width=500></div>
+<p style="text-align: center;font-style:italic">Table 1: Validation F1</p>
+
+* The performance of the student network has been validated as satisfactory and the multi-layer attention-based explanations are extracted for analysis
+
+&nbsp;
+
+## Explanation SOTA methods
+
+* ***Attention-based: Last-layer explainability (LL)*** The last layer’s self-attention head of the student’s encoder was analyzed. The best concept group to explain a given
+prediction was defined as that which corresponds to the highest attention value
+
+* ***Gradient-based: Saliency explainability (SA)*** The gradients of the loss function with respect to the input (concept groups) were computed. The best concept group to explain a given prediction was defined as that which yields the largest mean absolute value
+
+* ***Perturbation-based: Shapley additive explanations (SH)*** The SHAP value of each feature was computed. The best concept group was defined as that with the largest mean absolute SHAP value
+
+* ***Attention head aggregation by averaging (AVG)*** This approach employs the teacher network instead of the distilled student and aggregates the heads from each layer by averaging them. The best concept group to explain a given prediction was defined as in our proposed MLA model (graph + Dijkstra algorithm)
```

Comment:
```suggestion
* ***Attention head aggregation by averaging (AVG)*** This approach employs the teacher network instead of the distilled student and aggregates the heads from each layer by averaging them. The best concept group to explain a given prediction was defined as in the proposed MLA model (graph + Dijkstra algorithm)
```